apis/refs/v1beta1/computerefs.go

```diff
@@ -55,6 +66,7 @@ func ResolveComputeNetwork(ctx context.Context, reader client.Reader, src client
 			return nil, fmt.Errorf("cannot specify both name and external on computenetwork reference")
 		}
 
+		ref.External = fixStaleExternalFormat(ref.External)
 		tokens := strings.Split(ref.External, "/")
 		if len(tokens) == 5 && tokens[0] == "projects" && tokens[2] == "global" && tokens[3] == "networks" {
```

Comment:
Should we handle it here instead?  e.g.

external := ref.External
external = strings.TrimPrefix(external, "https://www.googleapis.com/compute/v1/")
external = strings.TrimPrefix(external, "https://www.googleapis.com/compute/v1beta1/")
tokens := strings.Split(external, "/")
		if len(tokens) == 5 && tokens[0] == "projects" && tokens[2] == "global" && tokens[3] == "networks" {
		```

Reply:
OTOH I do like the idea that you have a nice function here that we can share with ComputeNetwork, ComputeSubnetwork etc etc and that we can write-back to ref.External when we are following that pattern...

Reply:
So I searched and checked around, I *think* all Compute Beta resource allows selfLink as "External" value, some special case allows IP Address as External value as well. I think when migrating this to the service API dir, we could have a top level function to guarantee all resource allows the old External value.